uvicorn/supervisors/watchgodreload.py

```diff
@@ -41,26 +123,14 @@ def __init__(
         super().__init__(config, target, sockets)
         self.reloader_name = "watchgod"
         self.watchers = []
-        watch_dirs = {
-            Path(watch_dir).resolve()
-            for watch_dir in self.config.reload_dirs
-            if Path(watch_dir).is_dir()
-        }
-        watch_dirs_set = set(watch_dirs)
-
-        # remove directories that already have a parent watched, so that we don't have
-        # duplicated change events
-        for watch_dir in watch_dirs:
-            for compare_dir in watch_dirs:
-                if compare_dir is watch_dir:
-                    continue
-
-                if compare_dir in watch_dir.parents:
-                    watch_dirs_set.remove(watch_dir)
-
-        self.watch_dir_set = watch_dirs_set
-        for w in watch_dirs_set:
-            self.watchers.append(CustomWatcher(str(w)))
+        reload_dirs = []
+        for directory in config.reload_dirs:
+            if Path.cwd() not in directory.parents:
+                reload_dirs.append(directory)
+        if Path.cwd() not in reload_dirs:
+            reload_dirs.append(Path.cwd())
```

Comment:
I'm just wondering about those 2 lines because I was trying the branch on my ucivorn repo:
it has a dev folder I gitignored to contain all the dev app I'm loading to debug, and several dot files that are the result of mypy / pytest etc...
```
uvicorn
|--dev
|--.git
|--.idea
|--.pytest_cache
```
now if I do `dev.app:app --reload --log-level=trace --reload-dir=dev`
I get 
```
INFO:     Will watch for changes in these directories: ['/home/lotso/PycharmProjects/uvicorn/dev']
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
DEBUG:    WatchGodReload detected a new excluded dir '.github' in '/home/lotso/PycharmProjects/uvicorn'; Adding to exclude list.
DEBUG:    WatchGodReload detected a new excluded dir '.git' in '/home/lotso/PycharmProjects/uvicorn'; Adding to exclude list.
DEBUG:    WatchGodReload detected a new excluded dir '.eggs' in '/home/lotso/PycharmProjects/uvicorn'; Adding to exclude list.
DEBUG:    WatchGodReload detected a new excluded dir '.idea' in '/home/lotso/PycharmProjects/uvicorn'; Adding to exclude list.
DEBUG:    WatchGodReload detected a new excluded dir '.pytest_cache' in '/home/lotso/PycharmProjects/uvicorn'; Adding to exclude list.
INFO:     Started reloader process [1445703] using watchgod
```

so looking for changes in dev is ok to me, the unexpected here is the dot folders at the same level.
not sure if that's intended

Reply:
Yeah that's intended, but I see that it is not very intuitive. I'll add a check for the exclude warning.
What is happening is the following:

- We add the reload dir to folders we watch
- We need to add the cwd for folders that could be later found via an include patterns
- The cwd is scanned and foders that are excluded via the default pattern .* are found in cwd, that leads to this messages

I'll check if found folders are in the reload dirs before firing of this warning, that should clear up the unnecessary information.

Reply:
I've added the check, but it will still log for the default patterns (esp. hidden directories) in watched directories.
I think this is fine as it is DEBUG level output...